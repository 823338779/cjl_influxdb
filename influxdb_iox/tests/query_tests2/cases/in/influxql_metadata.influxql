-- IOX_SETUP: InfluxQLSelectSupport

-- SHOW FIELD KEYS
SHOW FIELD KEYS;
SHOW FIELD KEYS LIMIT 2;
SHOW FIELD KEYS OFFSET 1;
SHOW FIELD KEYS LIMIT 1 OFFSET 2;
SHOW FIELD KEYS FROM cpu;
SHOW FIELD KEYS FROM disk,cpu,disk;
SHOW FIELD KEYS FROM cpu,disk,cpu;
SHOW FIELD KEYS FROM /m.*/;
SHOW FIELD KEYS FROM /d\isk/;
SHOW FIELD KEYS FROM does_not_exist;

-- unimplemented features in `SHOW FIELD KEYS`
SHOW FIELD KEYS ON my_db;
SHOW FIELD KEYS FROM x.my_db;
SHOW FIELD KEYS FROM x.y.my_db;

-- SHOW TAG VALUES
-- Note: need to sort til https://github.com/influxdata/influxdb_iox/issues/7513 is fixed
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY = "tag0";
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY = "does_not_exist";
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY != "tag0";
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY != "does_not_exist";
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY =~ /tag.*/;
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY !~ /tag.*/;
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY IN ("device", "host", "device");
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY = "tag0" LIMIT 2;
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY = "tag0" OFFSET 1;
-- IOX_COMPARE: sorted
SHOW TAG VALUES WITH KEY = "tag0" LIMIT 1 OFFSET 2;
-- IOX_COMPARE: sorted
SHOW TAG VALUES FROM m0 WITH KEY = "tag0";
-- IOX_COMPARE: sorted
SHOW TAG VALUES FROM m0,m1,m0 WITH KEY = "tag0";
-- IOX_COMPARE: sorted
SHOW TAG VALUES FROM m1,m0,m1 WITH KEY = "tag0";
-- IOX_COMPARE: sorted
SHOW TAG VALUES FROM /m.*/ WITH KEY = "tag0";
-- IOX_COMPARE: sorted
SHOW TAG VALUES FROM /d\isk/ WITH KEY = "device";
-- IOX_COMPARE: sorted
SHOW TAG VALUES FROM does_not_exist WITH KEY = "tag0";

-- unimplemented features in `SHOW TAG VALUES`
SHOW TAG VALUES ON my_db WITH KEY = "tag0";
SHOW TAG VALUES FROM x.my_db WITH KEY = "tag0";
SHOW TAG VALUES FROM x.y.my_db WITH KEY = "tag0";
SHOW TAG VALUES WITH KEY = "tag0" WHERE tag1 = "foo";
