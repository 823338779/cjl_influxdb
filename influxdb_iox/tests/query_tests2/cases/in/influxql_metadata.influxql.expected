-- Test Setup: InfluxQLSelectSupport
-- InfluxQL: SHOW FIELD KEYS;
name: cpu
+--------------+-----------+
| fieldKey     | fieldType |
+--------------+-----------+
| usage_idle   | float     |
| usage_system | float     |
+--------------+-----------+
name: disk
+------------+-----------+
| fieldKey   | fieldType |
+------------+-----------+
| bytes_free | integer   |
| bytes_used | integer   |
+------------+-----------+
name: m0
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| f64      | float     |
| i64      | integer   |
| str      | string    |
+----------+-----------+
name: m1
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| f64      | float     |
| i64      | integer   |
| str      | string    |
+----------+-----------+
name: m2
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| f64      | float     |
+----------+-----------+
name: m3
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| u64      | unsigned  |
+----------+-----------+
name: m4
+-----------+-----------+
| fieldKey  | fieldType |
+-----------+-----------+
| field.one | float     |
+-----------+-----------+
-- InfluxQL: SHOW FIELD KEYS LIMIT 2;
name: cpu
+--------------+-----------+
| fieldKey     | fieldType |
+--------------+-----------+
| usage_idle   | float     |
| usage_system | float     |
+--------------+-----------+
name: disk
+------------+-----------+
| fieldKey   | fieldType |
+------------+-----------+
| bytes_free | integer   |
| bytes_used | integer   |
+------------+-----------+
name: m0
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| f64      | float     |
| i64      | integer   |
+----------+-----------+
name: m1
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| f64      | float     |
| i64      | integer   |
+----------+-----------+
name: m2
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| f64      | float     |
+----------+-----------+
name: m3
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| u64      | unsigned  |
+----------+-----------+
name: m4
+-----------+-----------+
| fieldKey  | fieldType |
+-----------+-----------+
| field.one | float     |
+-----------+-----------+
-- InfluxQL: SHOW FIELD KEYS OFFSET 1;
name: cpu
+--------------+-----------+
| fieldKey     | fieldType |
+--------------+-----------+
| usage_system | float     |
+--------------+-----------+
name: disk
+------------+-----------+
| fieldKey   | fieldType |
+------------+-----------+
| bytes_used | integer   |
+------------+-----------+
name: m0
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| i64      | integer   |
| str      | string    |
+----------+-----------+
name: m1
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| i64      | integer   |
| str      | string    |
+----------+-----------+
-- InfluxQL: SHOW FIELD KEYS LIMIT 1 OFFSET 2;
name: m0
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| str      | string    |
+----------+-----------+
name: m1
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| str      | string    |
+----------+-----------+
-- InfluxQL: SHOW FIELD KEYS FROM cpu;
name: cpu
+--------------+-----------+
| fieldKey     | fieldType |
+--------------+-----------+
| usage_idle   | float     |
| usage_system | float     |
+--------------+-----------+
-- InfluxQL: SHOW FIELD KEYS FROM disk,cpu,disk;
name: cpu
+--------------+-----------+
| fieldKey     | fieldType |
+--------------+-----------+
| usage_idle   | float     |
| usage_system | float     |
+--------------+-----------+
name: disk
+------------+-----------+
| fieldKey   | fieldType |
+------------+-----------+
| bytes_free | integer   |
| bytes_used | integer   |
+------------+-----------+
-- InfluxQL: SHOW FIELD KEYS FROM cpu,disk,cpu;
name: cpu
+--------------+-----------+
| fieldKey     | fieldType |
+--------------+-----------+
| usage_idle   | float     |
| usage_system | float     |
+--------------+-----------+
name: disk
+------------+-----------+
| fieldKey   | fieldType |
+------------+-----------+
| bytes_free | integer   |
| bytes_used | integer   |
+------------+-----------+
-- InfluxQL: SHOW FIELD KEYS FROM /m.*/;
name: m0
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| f64      | float     |
| i64      | integer   |
| str      | string    |
+----------+-----------+
name: m1
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| f64      | float     |
| i64      | integer   |
| str      | string    |
+----------+-----------+
name: m2
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| f64      | float     |
+----------+-----------+
name: m3
+----------+-----------+
| fieldKey | fieldType |
+----------+-----------+
| u64      | unsigned  |
+----------+-----------+
name: m4
+-----------+-----------+
| fieldKey  | fieldType |
+-----------+-----------+
| field.one | float     |
+-----------+-----------+
-- InfluxQL: SHOW FIELD KEYS FROM /d\isk/;
name: disk
+------------+-----------+
| fieldKey   | fieldType |
+------------+-----------+
| bytes_free | integer   |
| bytes_used | integer   |
+------------+-----------+
-- InfluxQL: SHOW FIELD KEYS FROM does_not_exist;
-- InfluxQL: SHOW FIELD KEYS ON my_db;
Error while planning query: This feature is not implemented: SHOW FIELD KEYS ON <database>
-- InfluxQL: SHOW FIELD KEYS FROM x.my_db;
Error while planning query: This feature is not implemented: retention policy in from clause
-- InfluxQL: SHOW FIELD KEYS FROM x.y.my_db;
Error while planning query: This feature is not implemented: database name in from clause
-- InfluxQL: SHOW TAG VALUES WITH KEY = "tag0";
-- Results After Sorting
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
+------+-------+
name: m1
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
+------+-------+
name: m2
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
| tag0 | val03 |
| tag0 | val04 |
| tag0 | val05 |
| tag0 | val07 |
| tag0 | val08 |
| tag0 | val09 |
| tag0 | val10 |
+------+-------+
-- InfluxQL: SHOW TAG VALUES WITH KEY = "does_not_exist";
-- Results After Sorting
-- InfluxQL: SHOW TAG VALUES WITH KEY != "tag0";
-- Results After Sorting
name: cpu
+------+-----------+
| key  | value     |
+------+-----------+
| cpu  | cpu-total |
| cpu  | cpu0      |
| cpu  | cpu1      |
| host | host1     |
+------+-----------+
name: disk
+--------+---------+
| key    | value   |
+--------+---------+
| device | disk1s1 |
| device | disk1s2 |
| device | disk1s5 |
| host   | host1   |
+--------+---------+
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag1 | val10 |
| tag1 |       |
+------+-------+
name: m4
+---------+-------+
| key     | value |
+---------+-------+
| tag.one | foo   |
+---------+-------+
-- InfluxQL: SHOW TAG VALUES WITH KEY != "does_not_exist";
-- Results After Sorting
name: cpu
+------+-----------+
| key  | value     |
+------+-----------+
| cpu  | cpu-total |
| cpu  | cpu0      |
| cpu  | cpu1      |
| host | host1     |
+------+-----------+
name: disk
+--------+---------+
| key    | value   |
+--------+---------+
| device | disk1s1 |
| device | disk1s2 |
| device | disk1s5 |
| host   | host1   |
+--------+---------+
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
| tag1 | val10 |
| tag1 |       |
+------+-------+
name: m1
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
+------+-------+
name: m2
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
| tag0 | val03 |
| tag0 | val04 |
| tag0 | val05 |
| tag0 | val07 |
| tag0 | val08 |
| tag0 | val09 |
| tag0 | val10 |
+------+-------+
name: m4
+---------+-------+
| key     | value |
+---------+-------+
| tag.one | foo   |
+---------+-------+
-- InfluxQL: SHOW TAG VALUES WITH KEY =~ /tag.*/;
-- Results After Sorting
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
| tag1 | val10 |
| tag1 |       |
+------+-------+
name: m1
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
+------+-------+
name: m2
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
| tag0 | val03 |
| tag0 | val04 |
| tag0 | val05 |
| tag0 | val07 |
| tag0 | val08 |
| tag0 | val09 |
| tag0 | val10 |
+------+-------+
name: m4
+---------+-------+
| key     | value |
+---------+-------+
| tag.one | foo   |
+---------+-------+
-- InfluxQL: SHOW TAG VALUES WITH KEY !~ /tag.*/;
-- Results After Sorting
name: cpu
+------+-----------+
| key  | value     |
+------+-----------+
| cpu  | cpu-total |
| cpu  | cpu0      |
| cpu  | cpu1      |
| host | host1     |
+------+-----------+
name: disk
+--------+---------+
| key    | value   |
+--------+---------+
| device | disk1s1 |
| device | disk1s2 |
| device | disk1s5 |
| host   | host1   |
+--------+---------+
-- InfluxQL: SHOW TAG VALUES WITH KEY IN ("device", "host", "device");
-- Results After Sorting
name: cpu
+------+-------+
| key  | value |
+------+-------+
| host | host1 |
+------+-------+
name: disk
+--------+---------+
| key    | value   |
+--------+---------+
| device | disk1s1 |
| device | disk1s2 |
| device | disk1s5 |
| host   | host1   |
+--------+---------+
-- InfluxQL: SHOW TAG VALUES WITH KEY = "tag0" LIMIT 2;
-- Results After Sorting
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
+------+-------+
name: m1
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
+------+-------+
name: m2
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
+------+-------+
-- InfluxQL: SHOW TAG VALUES WITH KEY = "tag0" OFFSET 1;
-- Results After Sorting
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val01 |
| tag0 | val02 |
+------+-------+
name: m1
+------+-------+
| key  | value |
+------+-------+
| tag0 | val01 |
+------+-------+
name: m2
+------+-------+
| key  | value |
+------+-------+
| tag0 | val01 |
| tag0 | val02 |
| tag0 | val03 |
| tag0 | val04 |
| tag0 | val05 |
| tag0 | val07 |
| tag0 | val08 |
| tag0 | val09 |
| tag0 | val10 |
+------+-------+
-- InfluxQL: SHOW TAG VALUES WITH KEY = "tag0" LIMIT 1 OFFSET 2;
-- Results After Sorting
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val02 |
+------+-------+
name: m2
+------+-------+
| key  | value |
+------+-------+
| tag0 | val02 |
+------+-------+
-- InfluxQL: SHOW TAG VALUES FROM m0 WITH KEY = "tag0";
-- Results After Sorting
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
+------+-------+
-- InfluxQL: SHOW TAG VALUES FROM m0,m1,m0 WITH KEY = "tag0";
-- Results After Sorting
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
+------+-------+
name: m1
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
+------+-------+
-- InfluxQL: SHOW TAG VALUES FROM m1,m0,m1 WITH KEY = "tag0";
-- Results After Sorting
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
+------+-------+
name: m1
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
+------+-------+
-- InfluxQL: SHOW TAG VALUES FROM /m.*/ WITH KEY = "tag0";
-- Results After Sorting
name: m0
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
+------+-------+
name: m1
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
+------+-------+
name: m2
+------+-------+
| key  | value |
+------+-------+
| tag0 | val00 |
| tag0 | val01 |
| tag0 | val02 |
| tag0 | val03 |
| tag0 | val04 |
| tag0 | val05 |
| tag0 | val07 |
| tag0 | val08 |
| tag0 | val09 |
| tag0 | val10 |
+------+-------+
-- InfluxQL: SHOW TAG VALUES FROM /d\isk/ WITH KEY = "device";
-- Results After Sorting
name: disk
+--------+---------+
| key    | value   |
+--------+---------+
| device | disk1s1 |
| device | disk1s2 |
| device | disk1s5 |
+--------+---------+
-- InfluxQL: SHOW TAG VALUES FROM does_not_exist WITH KEY = "tag0";
-- Results After Sorting
-- InfluxQL: SHOW TAG VALUES ON my_db WITH KEY = "tag0";
Error while planning query: This feature is not implemented: SHOW TAG VALUES ON <database>
-- InfluxQL: SHOW TAG VALUES FROM x.my_db WITH KEY = "tag0";
Error while planning query: This feature is not implemented: retention policy in from clause
-- InfluxQL: SHOW TAG VALUES FROM x.y.my_db WITH KEY = "tag0";
Error while planning query: This feature is not implemented: database name in from clause
-- InfluxQL: SHOW TAG VALUES WITH KEY = "tag0" WHERE tag1 = "foo";
Error while planning query: This feature is not implemented: SHOW TAG VALUES WHERE <condition>